### PR TITLE
Re add agent type definitions

### PR DESCRIPTION
**Pull request by Redent0r**
This PR re adds the Storage struct definition and replaces the need to use the autogenerated one from ttrpc.

We do this to address an issue with ttrpc when used on Windows

Test run: https://dev.azure.com/mariner-org/mariner/_build/results?buildId=463780&view=ms.vss-test-web.build-test-results-tab [100%]

### CODE REVIEW

File: src/tools/genpolicy/Cargo.toml [hunk 1]
```diff
@@ -45,7 +45,6 @@ tokio = {version = "1.33.0", features = ["rt-multi-thread"]}
 oci = { path = "../../libs/oci" }
 
 # Kata Agent prototol.
-protocols = { path = "../../libs/protocols", features = ["with-serde"] }
 protobuf = "3.2.0"
```
Comment (danmihai1) [inline src/tools/genpolicy/Cargo.toml:48]: Shouldn't protobuf be removed too?

File: src/tools/genpolicy/src/config_map.rs [hunk 2]
```diff
@@ -11,10 +11,10 @@ use crate::pod;
 use crate::policy;
 use crate::settings;
 use crate::yaml;
+use crate::agent;
```
Comment (danmihai1) [inline src/tools/genpolicy/src/config_map.rs:14]: Order alphabetically. 

If you use rust-analyzer from VSCode, that might order correctly for you.

**Review by danmihai1 — APPROVED** [review]

File: src/tools/genpolicy/Cargo.toml (hunk 1 above)
Comment (Redent0r) [reply src/tools/genpolicy/Cargo.toml]: Wasn't sure about this one. Removed. Thanks!

File: src/tools/genpolicy/src/config_map.rs (hunk 2 above)
Comment (Redent0r) [reply src/tools/genpolicy/src/config_map.rs]: I do have rust-analyser and its very helpful, but mine didn't complain about the aphabetical order of imports. I might be missing a setting or another extension. Updated. Thanks!

**Review by ms-mahuber — APPROVED** [review]

File: src/agent/samples/policy/yaml/deployment/deployment-busybox.yaml [hunk 3]
```diff
@@ -13,7 +13,7 @@ spec:
...
...JCIsCiAgICAgICAgIl4kKHN2Y19uYW1lKV9QT1JUXyQoaXBfcClfVENQPXRjcDovLyQoaXB2NF9hKTokKGlwX3ApJCIsCiAgICAgICAgIl4kKHN2Y19uYW1lKV9QT1JUXyQoaXBfcClfVENQX1BST1RPPXRjcCQiLAogICAgICAgICJeJChzdmNfbmFtZSlfUE9SVF8kKGlwX3ApX1RDUF9QT1JUPSQoaXBfcCkkIiwKICAgICAgICAiXiQoc3ZjX25hbWUpX1BPUlRfJChpcF9wKV9UQ1BfQUREUj0kKGlwdjRfYSkkIiwKICAgICAgICAiXiQoc3ZjX25hbWUpX1NFUlZJQ0VfSE9TVD0kKGlwdjRfYSkkIiwKICAgICAgICAiXiQoc3ZjX25hbWUpX1NFUlZJQ0VfUE9SVD0kKGlwX3ApJCIsCiAgICAgICAgIl4kKHN2Y19uYW1lKV9TRVJWSUNFX1BPUlRfJChkbnNfbGFiZWwpPSQoaXBfcCkkIiwKICAgICAgICAiXiQoc3ZjX25hbWUpX1BPUlQ9dGNwOi8vJChpcHY0X2EpOiQoaXBfcCkkIiwKICAgICAgICAiXkFaVVJFX0NMSUVOVF9JRD1bQS1GYS1mMC05LV0rJCIsCiAgICAgICAgIl5BWlVSRV9URU5BTlRfSUQ9W0EtRmEtZjAtOS1dKyQiLAogICAgICAgICJeQVpVUkVfRkVERVJBVEVEX1RPS0VOX0ZJTEU9L3Zhci9ydW4vc2VjcmV0cy9henVyZS90b2tlbnMvYXp1cmUtaWRlbnRpdHktdG9rZW4kIiwKICAgICAgICAiXkFaVVJFX0FVVEhPUklUWV9IT1NUPWh0dHBzOi8vbG9naW5cXC5taWNyb3NvZnRvbmxpbmVcXC5jb20vJCIKICAgICAgXQogICAgfSwKICAgICJDb3B5RmlsZVJlcXVlc3QiOiBbCiAgICAgICJeJChjcGF0aCkvIgogICAgXSwKICAgICJFeGVjUHJvY2Vzc1JlcXVlc3QiOiB7CiAgICAgICJjb21tYW5kcyI6IFtdLAogICAgICAicmVnZXgiOiBbXQogICAgfSwKICAgICJSZWFkU3RyZWFtUmVxdWVzdCI6IHRydWUsCiAgICAiV3JpdGVTdHJlYW1SZXF1ZXN0IjogZmFsc2UKICB9Cn0=
```
Comment (Redent0r) [inline src/agent/samples/policy/yaml/deployment/deployment-busybox.yaml:16]: This annotation needed to be updated in cc-msft-prototypes. I double checked and produces the exact annotation when generating from cc-msft-prototypes. So updating here

**Pull request merged**